src/android/interoperability/cpp/rust-result.md

```diff
@@ -0,0 +1,16 @@
+# Rust Error Handling
+
+```rust,ignore
+{{#include ../../../../third_party/cxx/book/snippets.rs:rust_result}}
+```
+
+<details>
+
+* Rust functions that return `Result` are translated to exceptions on the C++
+  side.
+* The exception thrown will always be of type `rust::Error`, which primarily
+  exposes a way to get the error message string.
```

Comment:
Is this via the Rust [`Error`](https://doc.rust-lang.org/stable/std/error/trait.Error.html) trait?

Reply:
Not sure, more likely it's using the `Display` impl though, since that's what you're supposed to use with the `std::Error` trait anyway.

Reply:
https://cxx.rs/binding/result.html#returning-result-from-rust-to-c indicates that this does use `Display`; we might note that here as well.

Reply:
Added a note about `Display`.